gprofiler/main.py

```diff
@@ -289,26 +306,40 @@ def _snapshot(self):
         except Exception:
             logger.error("Running perf failed; consider running gProfiler with '--perf-mode none' to avoid using perf")
             raise
-
+        metadata = (
+            get_current_metadata(self._static_metadata) if self._collect_metadata else {"hostname": get_hostname()}
+        )
         if self._runtimes["system"]:
             merged_result, total_samples = merge.merge_profiles(
                 system_result,
                 process_profiles,
                 self._docker_client,
+                metadata,
             )
         else:
             assert system_result == {}, system_result  # should be empty!
             merged_result, total_samples = merge.concatenate_profiles(
                 process_profiles,
                 self._docker_client,
+                metadata,
             )
 
         if self._output_dir:
             self._generate_output_files(merged_result, local_start_time, local_end_time)
 
         if self._client:
+            cpu_avg = self._system_metrics_monitor.get_cpu_utilization()
```

Comment:
Like I commented elsewhere - instead of 2 calls here, 1 call `get_metrics()` and pass it onward. That "metric" can be a dict you wanna send back.